controller/UndoManager.cpp

```diff
@@ -0,0 +1,38 @@
+//
+//  UndoManager.cpp
+//  demo
+//
+//  Created by Erik M. Buck on 3/16/21.
+//  Copyright © 2021 Erik M. Buck. All rights reserved.
+//
+
+#include "UndoManager.hpp"
+
+namespace WSU {
+namespace Controller {
+
+void UndoManager::undo() {
+    m_isUndoing = true;
```

Comment:
This `m_isUndoing` member variable is a bit odd because it creates action-at-a-distance. It is read only in `runCommand()` and would probably be better used as a local variable passed to `runCommand()` as a argument. Or better yet, for more readability and maintainability, a two-member `enum { UNDO, REDO }` as the argument rather than a boolean literal at the call site that requires reading the callee to understand.

Reply:
A feature of this design is that calling UndoManager::runCommand() always puts the reciprocal command on the correct stack, undo or redo. The caller doesn't need to know whether undo is in progress or not when runCommand() is called, and the correct result will happen anyway. Passing UNDO or REDO as a parameter requires the caller to know which is appropriate, and to know that, we would end up with some kind of if statement that makes decisions based on internal state of the UndoManager class at every single call site for runCommand().

Reply:
OK, that makes sense. Withdrawn.